smartcontract/service/native/auth/auth.go

```diff
@@ -0,0 +1,604 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package auth
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/ontio/ontology/common/serialization"
+	"github.com/ontio/ontology/core/genesis"
+	"github.com/ontio/ontology/errors"
+	. "github.com/ontio/ontology/smartcontract/service/native"
+	"github.com/ontio/ontology/smartcontract/service/native/utils"
+	"time"
+)
+
+var (
+	null = []byte{}
+)
+
+func Init() {
+	Contracts[genesis.AuthContractAddress] = RegisterAuthContract
+}
+
+func GetContractAdmin(native *NativeService, contractAddr []byte) ([]byte, error) {
+	key, err := GetContractAdminKey(native, contractAddr)
+	if err != nil {
+		return nil, err
+	}
+	item, err := utils.GetStorageItem(native, key)
+	if err != nil {
+		return nil, err
+	}
+	if item == nil {
+		return nil, nil
+	}
+	return item.Value, nil
+}
+
+/*
+ * contract admin management
+ */
+
+func initContractAdmin(native *NativeService, contractAddr, ontID []byte) (bool, error) {
+	admin, err := GetContractAdmin(native, contractAddr)
+	if err != nil {
+		return false, err
+	}
+	if admin != nil {
+		//admin is already set, just return
+		return false, nil
+	}
+
+	adminKey, err := GetContractAdminKey(native, contractAddr)
+	if err != nil {
+		return false, err
+	}
+	PutBytes(native, adminKey, ontID)
+	return true, nil
+}
+
+func InitContractAdmin(native *NativeService) ([]byte, error) {
+	param := new(InitContractAdminParam)
+	rd := bytes.NewReader(native.Input)
+	if err := param.Deserialize(rd); err != nil {
+		return nil, err
+	}
+
+	/*
+		cxt := native.ContextRef.CallingContext()
+		if cxt == nil {
+			return nil, errors.NewErr("no calling context")
+		}
+		invokeAddr := cxt.ContractAddress
+	*/
+
+	invokeAddr := genesis.OntContractAddress
```

Comment:
why OntContractAddress?

Reply:
sorry for this, should comment this line out and uncomment the four lines above.